x-pack/plugins/maps/public/shared/layers/sources/kibana_tilemap_source/kibana_tilemap_source.js

```diff
@@ -51,12 +52,19 @@ export class KibanaTilemapSource extends  AbstractTMSSource {
     });
   }
 
-
   async getUrlTemplate() {
-    return this._descriptor.url;
+    const tilemap = await getKibanaTileMap();
+    if (!tilemap || !tilemap.url) {
```

Comment:
No need to checkt for !tilemap since getKibanaTileMap will always return an empty state